src/widget.h

```diff
@@ -118,6 +137,24 @@ class generic_factory;
 // Forward declaration, due to codependency on panels.h
 class window_panel;
 
+struct widget_phrase {
```

Comment:
I'm glad to see you made this a separate structure, I was a little worried how messy it would get trying to parse them on the fly inside of loading widgets. The inclusion of `id` to uniquely identify phrases will give us a lot of power too.